backend/src/services/premium/enrichment/organizationEnrichmentService.ts

```diff
@@ -186,44 +184,30 @@ export default class OrganizationEnrichmentService extends LoggerBase {
 
         delete org.identities
 
-        // Check for an organization with the same website exists
-        let existingOrg
-        let orgService
-
+        // check for organization with same website and add them to merge suggestions
         if (org.website) {
-          existingOrg = await OrganizationRepository.findByDomain(org.website, this.options)
-          orgService = new OrganizationService({ ...this.options, transaction })
-        }
-
-        if (existingOrg && existingOrg.id !== org.id) {
-          await orgService.merge(existingOrg.id, org.id)
-          unmergedOrgs.push({ ...org, id: existingOrg.id })
-        } else {
-          unmergedOrgs.push(org)
-        }
-      }
+          let existingOrg = await OrganizationRepository.findByDomain(org.website, this.options)
 
-      // Check if two or more orgs in the umergedOrgs list have same website
-      const duplicateOrgs = []
-      const uniqueWebsites = new Set()
+          if (!existingOrg && existingOrgMap.has(org.website)) {
+            existingOrg = existingOrgMap.get(org.website)
+          }
```

Comment:
Shouldn't we do this the other way around? If it doesn't exist in the map, we ask the DB and set the map, if it exists in the map we shouldn't ask the db again unnecessarily - what do u think?